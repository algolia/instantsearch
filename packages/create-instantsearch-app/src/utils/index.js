const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const algoliasearch = require('algoliasearch');
const chalk = require('chalk');
const semver = require('semver');
const validateProjectName = require('validate-npm-package-name');

const TEMPLATES_FOLDER = path.join(__dirname, '../templates');

const algoliaConfig = {
  appId: 'OFCNCOG2CU',
  apiKey: 'f54e21fa3a2a0160595bb058179bfb1e',
  indexName: 'npm-search',
};

const client = algoliasearch(algoliaConfig.appId, algoliaConfig.apiKey);
const index = client.initIndex(algoliaConfig.indexName);

function checkAppName(appName) {
  const validationResult = validateProjectName(appName);

  if (!validationResult.validForNewPackages) {
    let errorMessage = `Could not create a project called "${chalk.red(
      appName
    )}" because of npm naming restrictions.`;

    (validationResult.errors || []).forEach((error) => {
      errorMessage += `\n  - ${error}`;
    });

    throw new Error(errorMessage);
  }

  return true;
}

function checkAppPath(appPath) {
  if (fs.existsSync(appPath)) {
    if (fs.lstatSync(appPath).isDirectory()) {
      const files = fs.readdirSync(appPath);

      if (files && files.length > 0) {
        throw new Error(
          `Could not create project in destination folder "${chalk.red(
            appPath
          )}" because it is not empty.`
        );
      }
    } else {
      throw new Error(
        `Could not create project at path ${chalk.red(
          appPath
        )} because a file of the same name already exists.`
      );
    }
  }

  if (!appPath) {
    throw new Error('Could not create project without directory');
  }

  return true;
}

function getAppTemplateConfig(templatePath, { loadFileFn = require } = {}) {
  try {
    return loadFileFn(path.join(templatePath, '.template.js'));
  } catch (err) {
    throw new Error(
      `The template configuration file \`.template.js\` contains errors:
${err.message}`
    );
  }
}

function isYarnAvailable() {
  try {
    execSync('yarnpkg --version', { stdio: 'ignore' });
    return true;
  } catch (err) {
    return false;
  }
}

function getAllTemplates() {
  const templates = fs
    .readdirSync(TEMPLATES_FOLDER)
    .map((name) => path.join(TEMPLATES_FOLDER, name))
    .filter((source) => fs.lstatSync(source).isDirectory())
    .map((source) => path.basename(source));

  return templates;
}

function getTemplatesByCategory() {
  const templatePaths = fs
    .readdirSync(TEMPLATES_FOLDER)
    .map((name) => path.join(TEMPLATES_FOLDER, name))
    .filter((source) => fs.lstatSync(source).isDirectory());

  const templates = templatePaths.reduce((allTemplates, source) => {
    const name = path.basename(source);

    const { category } = require(`${source}/.template.js`);

    if (!category) {
      return allTemplates;
    }

    return {
      ...allTemplates,
      [category]: [...(allTemplates[category] || []), name],
    };
  }, {});

  return templates;
}

function getTemplatePath(templateName) {
  const supportedTemplates = getAllTemplates();

  // We support the template, let's retrieve its path
  if (supportedTemplates.includes(templateName)) {
    return path.join(TEMPLATES_FOLDER, templateName);
  }

  // This is a custom template, it's a path already
  return templateName;
}

async function fetchLibraryVersions(libraryName) {
  const library = await index.getObject(libraryName, {
    attributesToRetrieve: ['versions'],
  });

  return Object.keys(library.versions).reverse();
}

function getLibraryVersion({ libraryName, supportedVersion = '' }) {
  return async (getVersion) => {
    const versions = await fetchLibraryVersions(libraryName);

    return getVersion(versions, supportedVersion);
  };
}

async function getLatestLibraryVersion(...args) {
  return await getLibraryVersion(...args)(semver.maxSatisfying);
}

async function getEarliestLibraryVersion(...args) {
  return await getLibraryVersion(...args)(semver.minSatisfying);
}

const splitArray = (string) =>
  string
    .split(',')
    .filter(Boolean)
    .map((x) => x.trim());

module.exports = {
  checkAppName,
  checkAppPath,
  getAppTemplateConfig,
  isYarnAvailable,
  fetchLibraryVersions,
  getLatestLibraryVersion,
  getEarliestLibraryVersion,
  getAllTemplates,
  getTemplatePath,
  getTemplatesByCategory,
  splitArray,
};
