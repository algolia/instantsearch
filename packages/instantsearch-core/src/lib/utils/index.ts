// This folder is used for utility functions that are useful within the InstantSearch Core library. Not for external usage. For that the `src/lib/public` folder should be used.

export * from './addAbsolutePosition';
export * from './addQueryID';
export * from './checkIndexUiState';
export * from './checkRendering';
export * from './clearRefinements';
export * from './createSendEventForFacet';
export * from './createSendEventForHits';
export * from './escapeFacetValue';
export * from './geo-search';
export * from './getAppIdAndApiKey';
export * from './getInsightsAnonymousToken';
export * from './getRefinements';
export * from './hydrateRecommendCache';
export * from './hydrateSearchClient';
export * from './mergeSearchParameters';
export * from './resolveSearchParameters';
export * from './setIndexHelperState';
