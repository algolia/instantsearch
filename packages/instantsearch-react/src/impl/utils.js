export function isSpecialClick(event) {
  const isMiddleClick = event.button === 1;
  return Boolean(
    isMiddleClick ||
    event.altKey ||
    event.ctrlKey ||
    event.metaKey ||
    event.shiftKey
  );
}

export function getTranslation(key, defaults, custom, ...params) {
  const translation =
    {}.hasOwnProperty.call(custom, key) ?
      custom[key] :
      defaults[key];
  if (typeof translation === 'function') {
    return translation(...params);
  }
  return translation;
}

export function capitalize(key) {
  return key.length === 0 ? '' : `${key[0].toUpperCase()}${key.slice(1)}`;
}
