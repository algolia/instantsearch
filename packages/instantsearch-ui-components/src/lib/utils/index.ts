export * from './find';
export * from './startsWith';
