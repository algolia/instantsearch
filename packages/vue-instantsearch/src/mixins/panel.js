import mitt from 'mitt';

import { isVue3 } from '../util/vue-compat';

export const PANEL_EMITTER_NAMESPACE = 'instantSearchPanelEmitter';
export const PANEL_CHANGE_EVENT = 'PANEL_CHANGE_EVENT';

export const createPanelProviderMixin = () => ({
  props: {
    emitter: {
      type: Object,
      required: false,
      default() {
        return mitt();
      },
    },
  },
  provide() {
    return {
      [PANEL_EMITTER_NAMESPACE]: this.emitter,
    };
  },
  data() {
    return {
      canRefine: true,
    };
  },
  created() {
    this.emitter.on(PANEL_CHANGE_EVENT, (value) => {
      this.updateCanRefine(value);
    });
  },
  [isVue3 ? 'beforeUnmount' : 'beforeDestroy']() {
    this.emitter.all.clear();
  },
  methods: {
    updateCanRefine(value) {
      this.canRefine = value;
    },
  },
});

export const createPanelConsumerMixin = ({
  mapStateToCanRefine = (state) => Boolean(state.canRefine),
} = {}) => ({
  inject: {
    emitter: {
      from: PANEL_EMITTER_NAMESPACE,
      default() {
        return {
          emit: () => {},
        };
      },
    },
  },
  data() {
    return {
      state: null,
      hasAlreadyEmitted: false,
    };
  },
  watch: {
    state: {
      immediate: true,
      handler(nextState, previousState) {
        if (!nextState) {
          return;
        }

        const previousCanRefine = mapStateToCanRefine(previousState || {});
        const nextCanRefine = mapStateToCanRefine(nextState);

        if (!this.hasAlreadyEmitted || previousCanRefine !== nextCanRefine) {
          this.emitter.emit(PANEL_CHANGE_EVENT, nextCanRefine);
          this.hasAlreadyEmitted = true;
        }
      },
    },
  },
});
