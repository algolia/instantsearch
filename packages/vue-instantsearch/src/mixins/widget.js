import { isVue3 } from '../util/vue-compat';
import { warn } from '../util/warn';

export const createWidgetMixin = (
  { connector } = {},
  additionalProperties = {}
) => ({
  inject: {
    instantSearchInstance: {
      from: '$_ais_instantSearchInstance',
      default() {
        const tag = this.$options._componentTag;
        throw new TypeError(
          `It looks like you forgot to wrap your Algolia search component "<${tag}>" inside of an "<ais-instant-search>" component.`
        );
      },
    },
    getParentIndex: {
      from: '$_ais_getParentIndex',
      default() {
        return () => this.instantSearchInstance.mainIndex;
      },
    },
  },
  data() {
    return {
      state: null,
    };
  },
  created() {
    if (typeof connector === 'function') {
      this.factory = connector(this.updateState, () => {});
      this.widget = {
        ...this.factory(this.widgetParams),
        ...additionalProperties,
      };
      this.getParentIndex().addWidgets([this.widget]);

      if (
        this.instantSearchInstance._initialResults &&
        !this.instantSearchInstance.started
      ) {
        if (typeof this.instantSearchInstance.__forceRender !== 'function') {
          throw new Error(
            'You are using server side rendering with <ais-instant-search> instead of <ais-instant-search-ssr>.'
          );
        }
        this.instantSearchInstance.__forceRender(
          this.widget,
          this.getParentIndex()
        );
      }
    } else if (connector !== true) {
      warn(
        `You are using the InstantSearch widget mixin, but didn't provide a connector.
While this is technically possible, and will give you access to the Helper,
it's not the recommended way of making custom components.

If you want to disable this message, pass { connector: true } to the mixin.

Read more on using connectors: https://alg.li/vue-custom`
      );
    }
  },
  [isVue3 ? 'beforeUnmount' : 'beforeDestroy']() {
    if (this.widget) {
      this.getParentIndex().removeWidgets([this.widget]);
    }
  },
  watch: {
    widgetParams: {
      handler(nextWidgetParams) {
        this.state = null;
        this.getParentIndex().removeWidgets([this.widget]);
        this.widget = {
          ...this.factory(nextWidgetParams),
          ...additionalProperties,
        };
        this.getParentIndex().addWidgets([this.widget]);
      },
      deep: true,
    },
  },
  methods: {
    updateState(state = {}, isFirstRender) {
      if (!isFirstRender) {
        // Avoid updating the state on first render
        // otherwise there will be a flash of placeholder data
        this.state = state;
      }
    },
  },
});
