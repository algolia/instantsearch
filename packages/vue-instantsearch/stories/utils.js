import { liteClient as algoliasearch } from 'algoliasearch/lite';

export const previewWrapper =
  ({
    searchClient = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76'),
    insightsClient,
    indexName = 'instant_search',
    hits = `
    <template v-slot="{ items }">
      <ol class="playground-hits">
        <li
          v-for="item in items"
          :key="item.objectID"
          class="playground-hits-item"
        >
          <div
            class="playground-hits-image"
            :style="{ backgroundImage: 'url(' + item.image + ')' }"
          />
          <div class="playground-hits-desc">
            <p>
              <ais-highlight attribute="name" :hit="item" />
            </p>
            <p>Rating: {{ item.rating }}âœ­</p>
            <p>Price: {{ item.price }}$</p>
          </div>
        </li>
      </ol>
    </template>
  `,
    filters = `
    <ais-refinement-list attribute="brand" />
    <ais-configure :hitsPerPage="3" />
  `,
    routing,
  } = {}) =>
  () => ({
    template: `
    <ais-instant-search
      :search-client="searchClient"
      index-name="${indexName}"
      :routing="routing"
      :insights-client="insightsClient"
    >
      <div class="vis-container vis-container-preview">
        <story />
      </div>

      <div class="vis-container vis-container-playground">
        <div class="panel-left">
          ${filters}
        </div>
        <div class="panel-right">
          <ais-search-box />
          <ais-stats />
          <ais-hits>
            ${hits}
          </ais-hits>
          <ais-pagination />
        </div>
      </div>
    </ais-instant-search>
  `,
    data() {
      return {
        searchClient,
        routing,
        insightsClient,
      };
    },
  });
