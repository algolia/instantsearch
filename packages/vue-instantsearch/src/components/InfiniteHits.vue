<template>
  <div v-if="state" :class="suit()">
    <slot
      v-if="showPrevious"
      name="loadPrevious"
      :refine-previous="refinePrevious"
      :page="state.results.page"
      :is-first-page="state.isFirstPage"
    >
      <button
        :class="[
          suit('loadPrevious'),
          state.isFirstPage && suit('loadPrevious', 'disabled'),
        ]"
        :disabled="state.isFirstPage"
        @click="refinePrevious()"
      >
        Show previous results
      </button>
    </slot>

    <slot
      :items="state.items"
      :results="state.results"
      :banner="state.banner"
      :is-last-page="state.isLastPage"
      :refine-previous="refinePrevious"
      :refine-next="refineNext"
      :refine="refineNext"
      :insights="state.insights"
      :send-event="state.sendEvent"
    >
      <template
        v-if="showBanner && state.banner && state.banner.image.urls[0].url"
      >
        <slot name="banner" :banner="state.banner">
          <aside :class="suit('banner')">
            <a
              v-if="state.banner.link"
              :href="state.banner.link.url"
              :target="state.banner.link.target"
              :class="suit('banner-link')"
            >
              <img
                :src="state.banner.image.urls[0].url"
                :alt="state.banner.image.title"
                :class="suit('banner-image')"
              />
            </a>
            <img
              v-else
              :src="state.banner.image.urls[0].url"
              :alt="state.banner.image.title"
              :class="suit('banner-image')"
            />
          </aside>
        </slot>
      </template>
      <ol :class="suit('list')">
        <li
          v-for="(item, index) in state.items"
          :class="suit('item')"
          :key="item.objectID"
          @click="state.sendEvent('click:internal', item, 'Hit Clicked')"
          @auxclick="state.sendEvent('click:internal', item, 'Hit Clicked')"
        >
          <slot
            name="item"
            :item="item"
            :index="index"
            :insights="state.insights"
            :send-event="state.sendEvent"
          >
            objectID: {{ item.objectID }}, index: {{ index }}
          </slot>
        </li>
      </ol>

      <slot
        name="loadMore"
        :refine-next="refineNext"
        :refine="refineNext"
        :page="state.results.page"
        :is-last-page="state.isLastPage"
      >
        <button
          :class="[
            suit('loadMore'),
            state.isLastPage && suit('loadMore', 'disabled'),
          ]"
          :disabled="state.isLastPage"
          @click="refineNext()"
        >
          Show more results
        </button>
      </slot>
    </slot>
  </div>
</template>

<script>
import { connectInfiniteHitsWithInsights } from 'instantsearch.js/es/connectors';

import { createSuitMixin } from '../mixins/suit';
import { createWidgetMixin } from '../mixins/widget';

export default {
  name: 'AisInfiniteHits',
  mixins: [
    createWidgetMixin(
      {
        connector: connectInfiniteHitsWithInsights,
      },
      {
        $$widgetType: 'ais.infiniteHits',
      }
    ),
    createSuitMixin({ name: 'InfiniteHits' }),
  ],
  props: {
    showBanner: {
      type: Boolean,
      default: true,
    },
    showPrevious: {
      type: Boolean,
      default: false,
    },
    escapeHTML: {
      type: Boolean,
      default: true,
    },
    transformItems: {
      type: Function,
      default: undefined,
    },
    cache: {
      type: Object,
      default: undefined,
    },
  },
  computed: {
    widgetParams() {
      return {
        showBanner: this.showBanner,
        showPrevious: this.showPrevious,
        escapeHTML: this.escapeHTML,
        transformItems: this.transformItems,
        cache: this.cache,
      };
    },
  },
  methods: {
    refinePrevious() {
      this.state.showPrevious();
    },
    refineNext() {
      this.state.showMore();
    },
  },
};
</script>
