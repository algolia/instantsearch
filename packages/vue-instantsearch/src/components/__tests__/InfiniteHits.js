/**
 * @jest-environment jsdom
 */

import { mount } from '../../../test/utils';
import { __setState } from '../../mixins/widget';
import InfiniteHits from '../InfiniteHits.vue';
import '../../../test/utils/sortedHtmlSerializer';

jest.mock('../../mixins/widget');

const defaultState = {
  widgetParams: {
    showPrevious: false,
    escapeHTML: true,
    transformItems: (items) => items,
  },
  sendEvent: jest.fn(),
  hits: [
    {
      objectID: '00001',
    },
    {
      objectID: '00002',
    },
    {
      objectID: '00003',
    },
    {
      objectID: '00004',
    },
    {
      objectID: '00005',
    },
  ],
  results: {
    page: 0,
  },
  isFirstPage: false,
  isLastPage: false,
  showMore: () => {},
  showPrevious: () => {},
};

it('accepts a escapeHTML prop', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(InfiniteHits, {
    propsData: {
      escapeHTML: true,
    },
  });

  expect(wrapper.vm.widgetParams.escapeHTML).toBe(true);
});

it('accepts a transformItems prop', () => {
  __setState({
    ...defaultState,
  });

  const transformItems = () => {};

  const wrapper = mount(InfiniteHits, {
    propsData: {
      transformItems,
    },
  });

  expect(wrapper.vm.widgetParams.transformItems).toBe(transformItems);
});

it('accepts a showPrevious prop', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(InfiniteHits, {
    propsData: {
      showPrevious: true,
    },
  });

  expect(wrapper.vm.widgetParams.showPrevious).toBe(true);
});

it('accepts a cache prop', () => {
  __setState({
    ...defaultState,
  });

  const cache = {
    read: () => {},
    write: () => {},
  };

  const wrapper = mount(InfiniteHits, {
    propsData: {
      cache,
    },
  });

  expect(wrapper.vm.widgetParams.cache).toEqual(cache);
});

it('renders correctly', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(InfiniteHits);

  expect(wrapper.html()).toMatchSnapshot();
});

it('renders correctly with a custom rendering', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount({
    components: { InfiniteHits },
    template: `
      <InfiniteHits>
        <template v-slot="{ items }">
          <div>
            <div v-for="item in items">
              {{item.objectID}}
            </div>
          </div>
        </template>
      </InfiniteHits>
    `,
  });

  expect(wrapper.html()).toMatchSnapshot();
});

it('renders correctly with a custom item rendering', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount({
    components: { InfiniteHits },
    template: `
      <InfiniteHits>
        <template v-slot:item="{ item }">
          <div>
            {{item.objectID}}
          </div>
        </template>
      </InfiniteHits>
    `,
  });

  expect(wrapper.html()).toMatchSnapshot();
});

it('renders correctly on the first page', () => {
  __setState({
    ...defaultState,
    isFirstPage: true,
  });

  const wrapper = mount(InfiniteHits, {
    propsData: {
      showPrevious: true,
    },
  });

  const previousButton = wrapper.find('.ais-InfiniteHits-loadPrevious');

  expect(previousButton.exists()).toEqual(true);
  expect(
    previousButton.classes('ais-InfiniteHits-loadPrevious--disabled')
  ).toEqual(true);
  expect(previousButton).vueToBeDisabled();
  expect(wrapper.htmlCompat()).toMatchSnapshot();
});

it('renders correctly on the last page', () => {
  __setState({
    ...defaultState,
    isLastPage: true,
  });

  const wrapper = mount(InfiniteHits);

  expect(wrapper.htmlCompat()).toMatchSnapshot();
});

it('renders correctly when not on the first page', () => {
  __setState({
    ...defaultState,
    isFirstPage: false,
    isLastPage: false,
  });

  const wrapper = mount(InfiniteHits, {
    propsData: {
      showPrevious: true,
    },
  });

  const previousButton = wrapper.find('.ais-InfiniteHits-loadPrevious');

  expect(previousButton.exists()).toEqual(true);
  expect(
    previousButton.classes('ais-InfiniteHits-loadPrevious--disabled')
  ).toEqual(false);
  expect(previousButton.attributes('disabled')).toEqual(undefined);
  expect(wrapper.html()).toMatchSnapshot();
});

it('expect to call showPrevious on click', async () => {
  const showPrevious = jest.fn();

  __setState({
    ...defaultState,
    showPrevious,
  });

  const wrapper = mount(InfiniteHits, {
    propsData: {
      showPrevious: true,
    },
  });

  expect(showPrevious).toHaveBeenCalledTimes(0);

  await wrapper.find('.ais-InfiniteHits-loadPrevious').trigger('click');

  expect(showPrevious).toHaveBeenCalledTimes(1);
});

it('expect to call showMore on click', async () => {
  const showMore = jest.fn();

  __setState({
    ...defaultState,
    showMore,
  });

  const wrapper = mount(InfiniteHits);

  expect(showMore).not.toHaveBeenCalled();

  await wrapper.find('.ais-InfiniteHits-loadMore').trigger('click');

  expect(showMore).toHaveBeenCalled();
});

it('exposes insights prop to the default slot', async () => {
  const insights = jest.fn();

  __setState({
    ...defaultState,
    insights,
  });

  const wrapper = mount({
    components: { InfiniteHits },
    template: `
      <InfiniteHits>
        <template v-slot="{ items, insights }">
        <ul>
        <li v-for="(item, itemIndex) in items" >
          <button :id="'add-to-cart-' + item.objectID" @click="insights('clickedObjectIDsAfterSearch', {eventName: 'Add to cart', objectIDs: [item.objectID]})">
            Add to cart
          </button>
        </li>
      </ul>
        </template>
      </InfiniteHits>
    `,
  });

  await wrapper.find('#add-to-cart-00002').trigger('click');
  expect(insights).toHaveBeenCalledWith('clickedObjectIDsAfterSearch', {
    eventName: 'Add to cart',
    objectIDs: ['00002'],
  });
});

it('exposes insights prop to the item slot', async () => {
  const insights = jest.fn();

  __setState({
    ...defaultState,
    insights,
  });

  const wrapper = mount({
    components: { InfiniteHits },
    template: `
      <InfiniteHits>
        <template v-slot:item="{ item, insights }">
          <div>
            <button :id="'add-to-cart-' + item.objectID" @click="insights('clickedObjectIDsAfterSearch', {eventName: 'Add to cart', objectIDs: [item.objectID]})">
              Add to cart
            </button>
          </div>
        </template>
      </InfiniteHits>
    `,
  });

  await wrapper.find('#add-to-cart-00002').trigger('click');
  expect(insights).toHaveBeenCalledWith('clickedObjectIDsAfterSearch', {
    eventName: 'Add to cart',
    objectIDs: ['00002'],
  });
});

it('exposes send-event method for insights middleware', async () => {
  const sendEvent = jest.fn();
  __setState({
    ...defaultState,
    sendEvent,
  });

  const wrapper = mount({
    components: { InfiniteHits },
    template: `
      <InfiniteHits>
        <template v-slot="{ sendEvent }">
          <div>
            <button @click="sendEvent()">Send Event</button>
          </div>
        </template>
      </InfiniteHits>
    `,
  });

  await wrapper.find('button').trigger('click');
  expect(sendEvent).toHaveBeenCalledTimes(1);
});
