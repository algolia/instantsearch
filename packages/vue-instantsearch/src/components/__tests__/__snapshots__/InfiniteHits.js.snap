// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render a banner when showBanner is false 1`] = `
<div class="ais-InfiniteHits">
  <aside class="ais-InfiniteHits-banner">
    <a class="ais-InfiniteHits-banner-link"
       href="https://www.algolia.com"
    >
      <img class="ais-InfiniteHits-banner-image"
           src="https://via.placeholder.com/550x250"
      >
    </a>
  </aside>
  <ol class="ais-InfiniteHits-list">
    <li class="ais-InfiniteHits-item">
      objectID: 00001, index: 0
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00002, index: 1
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00003, index: 2
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00004, index: 3
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00005, index: 4
    </li>
  </ol>
  <button class="ais-InfiniteHits-loadMore">
    Show more results
  </button>
</div>
`;

exports[`renders correctly with a custom item rendering 1`] = `
<div class="ais-InfiniteHits">
  <ol class="ais-InfiniteHits-list">
    <li class="ais-InfiniteHits-item">
      <div>
        00001
      </div>
    </li>
    <li class="ais-InfiniteHits-item">
      <div>
        00002
      </div>
    </li>
    <li class="ais-InfiniteHits-item">
      <div>
        00003
      </div>
    </li>
    <li class="ais-InfiniteHits-item">
      <div>
        00004
      </div>
    </li>
    <li class="ais-InfiniteHits-item">
      <div>
        00005
      </div>
    </li>
  </ol>
  <button class="ais-InfiniteHits-loadMore">
    Show more results
  </button>
</div>
`;

exports[`renders correctly with a custom rendering 1`] = `
<div class="ais-InfiniteHits">
  <div>
    <div>
      00001
    </div>
    <div>
      00002
    </div>
    <div>
      00003
    </div>
    <div>
      00004
    </div>
    <div>
      00005
    </div>
  </div>
</div>
`;

exports[`renders correctly with banner data 1`] = `
<div class="ais-InfiniteHits">
  <aside class="ais-InfiniteHits-banner">
    <a class="ais-InfiniteHits-banner-link"
       href="https://www.algolia.com"
    >
      <img class="ais-InfiniteHits-banner-image"
           src="https://via.placeholder.com/550x250"
      >
    </a>
  </aside>
  <ol class="ais-InfiniteHits-list">
    <li class="ais-InfiniteHits-item">
      objectID: 00001, index: 0
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00002, index: 1
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00003, index: 2
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00004, index: 3
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00005, index: 4
    </li>
  </ol>
  <button class="ais-InfiniteHits-loadMore">
    Show more results
  </button>
</div>
`;

exports[`renders correctly with banner data and custom banner rendering 1`] = `
<div class="ais-InfiniteHits">
  <img src="https://via.placeholder.com/550x250">
  <ol class="ais-InfiniteHits-list">
    <li class="ais-InfiniteHits-item">
      objectID: 00001, index: 0
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00002, index: 1
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00003, index: 2
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00004, index: 3
    </li>
    <li class="ais-InfiniteHits-item">
      objectID: 00005, index: 4
    </li>
  </ol>
  <button class="ais-InfiniteHits-loadMore">
    Show more results
  </button>
</div>
`;
